multi-currency/src/Compatibility/WooCommerceSubscriptions.php

```diff
@@ -235,6 +235,11 @@ public function should_convert_product_price( bool $return, $product ): bool {
 			return $return;
 		}
 
+		// If product is not a subscription then bail.
+		if ( ! \WC_Subscriptions_Product::is_subscription( $product ) ) {
+			return $return;
+		}
```

Comment:
@jessepearson this is a first attempt at fixing #9458,  so it may not be the final code. While I’m checking to ensure it doesn’t interfere with any cases from your PR #7092, could you take a quick look to see if this approach seems correct or if I’m overlooking something obvious?

The bug is that this filter is being applied to non-subscription products when paying for a subscription renewal (specifically in the cart page). I think this filter should only be applied to subscriptions. What do you think?

Reply:
I've tested the first couple cases listed in the PR so far and found no issues. I'll continue testing later.

Reply:
I haven't tested this yet, however, I do have a question... should we be allowing other products to be added to the cart during a subscription renewal? I thought that WCSubs had something in place that would remove items from the cart in instances like this.

Reply:
@jessepearson hmm, I hadn't seen that restriction before until I was testing the case "Subscription switching testing - variable product" from #7092 where the message `A subscription has been removed from your cart. Products and subscriptions cannot be purchased at the same time.` appeared. The subscription was removed from the cart, leaving only the simple product that was additionally added to the cart.

Should this behavior be considered for this issue as well? If that's the case, the fix I'm proposing in this PR would become invalid.

@bborman22, who can confirm this behavior? So far, the fix I’m applying in this PR resolves the issue, and I haven’t encountered any problems in my tests. But perhaps it would be better not to allow products to be added when paying for a subscription renewal?

I wonder how long GS has been testing the case "[Manual renewal order testing](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-8.3.0#manual-renewal-order-testing)"? Since it seems to have been broken for quite some time.

Reply:
@bborman22 I've noticed that although it's allowed to add a product to the cart while paying for a subscription renewal, the product isn't considered in the order - only the subscription is paid. So the fix I’m making here isn’t very useful, since additional products aren't considered apart from the subscription.

I think we should: 
- Discard my fix.
- Update the PR to not allow adding additional products when paying for a subscription renewal. (@diegocurbelo Could someone on your team confirm if this is the expected behavior?)
- Update the  [Manual renewal order testing](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-8.3.0#manual-renewal-order-testing)  test case to verify that additional products CANNOT be added when paying for a subscription renewal.

Reply:
FYI: I went ahead and prepared a PR to prevent adding products when renewing a subscription. https://github.com/Automattic/woocommerce-subscriptions-core/pull/714

Reply:
@bborman22 @diegocurbelo I have updated https://github.com/Automattic/woocommerce-subscriptions-core/pull/714, and it should be ready for review if that is indeed the expected behavior when a subscription renewal is present in the cart. (I am inclined to believe it’s correct since currently adding another subscription is not allowed; I have simply made it so that adding any other product is also not permitted.)

Brian, note that if this is the case, I am not fixing the conversion issue; otherwise, we will never observe that case again. I would need to update the “[Manual renewal order testing](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-8.3.0#manual-renewal-order-testing)” to verify that adding a product to the cart is not allowed.

Please let me know if my PR in `woocommerce-subscriptions-core` is correct, as I will be out of the office from tomorrow, Wednesday, until Monday.

Reply:
@bborman22 the original issue aimed to fix a conversion bug found during testing for WooPayments 8.3. These test instructions are not part of the critical flows. Therefore, should we add the ‘Manual renewal order testing’ instructions again for the next release 8.5? What do you think? Or is it unnecessary, since technically the change was made outside of WooPayments?

Reply:
I think we could provide an updated test instructions for one more release (8.6) to confirm that the new flow is working as expected as it does address a previous issue within WooPayments. I figure one extra test for one cycle shouldn't be too costly and err on the side of caution / confirmation.